primitives/src/signer.rs

```diff
@@ -90,3 +90,37 @@ where
 		self.extrinsic_address.clone()
 	}
 }
+
+impl<Signer, Runtime> From<Signer> for ExtrinsicSigner<Signer, Signer::Signature, Runtime>
+where
+	Signer: Pair,
+	Runtime: FrameSystemConfig,
+	Runtime::AccountId: From<Signer::Public>,
+{
+	fn from(value: Signer) -> Self {
+		ExtrinsicSigner::<Signer, Signer::Signature, Runtime>::new(value)
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use crate::ExtrinsicSigner;
+	use kitchensink_runtime::Runtime;
+	use sp_core::{sr25519, Pair};
+
+	#[test]
+	fn test_extrinsic_signer_from_sr25519_pair() {
+		let alice: sr25519::Pair = Pair::from_string(
+			"0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a",
+			None,
+		)
+		.unwrap();
+
+		let es_from =
+			ExtrinsicSigner::<sr25519::Pair, sr25519::Signature, Runtime>::from(alice.clone());
```

Comment:
I don't think that's what was imagined in #452. The cool thing about `From` is that a simple `.into()` does everything for you, no need to call any complex function.

An example can be found here, where the public address is simply converted to a Runtime::Account:
https://github.com/scs/substrate-api-client/blob/027ed593d0907255c2c2460ed9b49ad3f7205143/primitives/src/signer.rs#L61

However, I do not know if that's even possible.

Reply:
Thanks, makes sense! (... still learning, so wasn't aware of that ...)